apps/meteor/app/livechat/server/lib/LivechatTyped.ts

```diff
@@ -459,6 +460,35 @@ class LivechatClass {
 			extraData,
 		});
 
+		if (isSingleContactEnabled()) {
+			let { contactId } = visitor;
+			if (contactId) {
+				const visitorContact = await LivechatVisitors.findOne<Pick<ILivechatVisitor, 'contactId'>>(visitor._id, {
+					projection: { contactId: 1 },
+				});
+				contactId = visitorContact?.contactId;
+			}
+
+			const contact = await LivechatContacts.findOne({ _id: contactId });
+
+			if (contact) {
+				const channel = contact.channels?.find((channel) => channel.name === roomInfo.source?.type && channel.visitorId === visitor._id);
+
+				if (!channel) {
+					Livechat.logger.debug(`Adding channel for contact ${contact._id}`);
+
+					await LivechatContacts.addChannel(contact._id, {
+						name: roomInfo.source?.label || roomInfo.source?.type.toString() || 'other',
+						visitorId: visitor._id,
+						blocked: false,
```

Comment:
Pretty curious why a channel can be blocked instead of just blocking the contact on all channels 👀

Reply:
It is by product design. Renato can explain it to you better, but the idea is that if your phone is stolen or your email is hacked, for example, the agent can block only the channel that is no longer trustworthy, and you can still contact the workspace by other means.

Reply:
thanks!